Review Update 5 breaking changes (#6684)
ballerina-platform/ballerina-dev-website: downloads/swan-lake-release-notes/swan-lake-2201.5.0 /RELEASE_NOTE.md

    ```ballerina
    public function main() {
        "abc"|string:Char a = "a";

>         var b = a + 1; // Compilation error now.

Author: Better to update if the suggestions improve the understandability.

    public function main() {
        json & readonly v = {};
        
        string|error r = v; // Compilation error now.
        Employee|string s = v.cloneReadOnly(); // Compilation error now.
    }
    ```

> - Fixed a bug that previously resulted in variables that were initialized with non-isolated expressions being inferred as `isolated` variables.

Reviewer: Why can't we use "to be"?

Author: Since `being` is also used, the `to be` preposition causes a repetition.

    ```ballerina
    int[] config = [];

    // `configs` was previously inferred as an `isolated` variable
    // incorrectly. It will no longer be inferred as an `isolated` 
    // variable since `config` is not an isolated expression.
    int[][] configs = [[1, 2], config]; 

    // Since `configs` is not inferred as an `isolated` variable now,
    // `getConfig` is not inferred as an `isolated` function.
    function getConfig(int index) returns int[] {
        lock {
            return configs[index].clone();
        }
    }
    ```

> - Fixed a bug (in dependently-typed function analysis) that previously resulted in compilation errors not being logged, when the `typedesc` argument is defined using a type definition (`T`) and the return type is a union (`T|t`) (where the basic types for `T` and `t` are not disjoint).

Reviewer: Why are we using parentheses for dependently-typed function analysis? It's a key part of the sentence without which the typedesc stuff don't make sense. Similarly for the last bit.

Reviewer: ```suggestion
- Fixed a bug in dependently-typed function analysis. Previously compilation errors were not logged when the `typedesc` argument is defined using a type definition (`T`) and the return type is a union (`T|t`) in which basic types for `T` and `t` are not disjoint.
```

Author: Need to use parentheses here because we need to have the word `that` right after the noun being described to avoid the meaning being distorted. The clause `that previously resulted in compilation errors not being logged` should be placed right after the noun being described by it.

Reviewer: Can you check the version I've suggested? I'd rather use that.

    }
    ```

> - Fixed a bug that resulted in inconsistent error messages associated with the `cloneWithType` operation.

Reviewer: ```suggestion
- Fixed a bug that resulted in inconsistent error messages when using the `value:cloneWithType` function.
```
